src/snowflake/snowpark/modin/plugin/_internal/join_utils.py

```diff
@@ -1192,10 +1212,11 @@ def align(
     # join columns of left and right matches, then there is no need to coalesce the join
     # keys, simply inherent from left gives the correct result.
     # Retaining the original columns also helps avoid unnecessary join in later steps.
-    if (
-        how == "outer"
-        and aligned_ordered_frame.ordering_columns != left.ordering_columns
-    ):
+    # if (
+    #    how == "outer"
```

Comment:
This seems a duplication with the current optimization. remove it for testing to see anything fails